clang_delta/EmptyStructToInt.cpp

```diff
@@ -213,6 +259,54 @@ void EmptyStructToInt::HandleTranslationUnit(ASTContext &Ctx)
     TransError = TransInternalError;
 }
 
+void EmptyStructToInt::handleOneRecordDecl(const RecordDecl *RD,
+                                           const RecordDecl *BaseRD,
+                                           const FieldDecl *FD,
+                                           unsigned int Idx)
+{
+  // Skip field if it is not of the type of one of
+  // the structs that are affected by the change
+  // Works recursively through chained structs
+  if (!RecordDeclToField[BaseRD] && BaseRD != TheRecordDecl) {
```

Comment:
I think it is needed for structs which are nested more than one level. For example:
```
struct S0 {
    int f;
};

struct S1 {
    int a;
    struct S0 s;
};

struct S2 {
    int a;
    struct S1 s;
    int b;
};

struct S2 s2 = {1, {2, {3}}, 4};
```

With the check the initialiser list "{3}" for `struct S0` in `struct S1` in `struct S2` in correctly changed to a zero value:
```


struct S1 {
    int a;
     int s;
};

struct S2 {
    int a;
    struct S1 s;
    int b;
};

struct S2 s2 = {1, {2, 0}, 4};
```

Without the check the initialiser list for `struct S0` is left unchanged:
```


struct S1 {
    int a;
     int s;
};

struct S2 {
    int a;
    struct S1 s;
    int b;
};

struct S2 s2 = {1, {2, {3}}, 4};
```

Although `struct S2` is not `TheRecordDecl` its initialiser list has to be changed since it contains a struct of the type under change.